pipeline_steps/common.groovy

```diff
@@ -1143,7 +1147,7 @@ from shlex import quote
 
 # We don't want to set a default for build_args as these do not always apply
 build_args = environ.get('SLAVE_CONTAINER_DOCKERFILE_BUILD_ARGS')
-dockerfile = environ.get('SLAVE_CONTAINER_DOCKERFILE_PATH', './Dockerfile')
+dockerfile = environ.get('SLAVE_CONTAINER_DOCKERFILE_PATH')
```

Comment:
What happens if the env var isn't set? What does a build arg of `-f ''` do? It might be worth putting in something more explicit here.

Reply:
Passing in `-f ''` seems to just assume a filename of `Dockerfile`, so I may as well just leave the default here to avoid confusion.

Reply:
I still don't think you want the default there, why would this function be called without `SLAVE_CONTAINER_DOCKERFILE_PATH` being set? If it happened it would be a bug, I'd rather we find out about it with an explicit failure than guess the name of the dockerfile.

Reply:
If we want to be explicit and make specifying `SLAVE_CONTAINER_DOCKERFILE_PATH` an absolute requirement then I can fail if that hasn't been set.  However, would the same not apply to `SLAVE_CONTAINER_DOCKERFILE_BUILD_ARGS`?

Reply:
That one is slightly different because there is no requirement for additional args but we could distinguish been the env var not being set and no args being specified null/''.

We've been caught out by defaults enough times before that I would personally always prefer have as few defaults nestled in the code as possible and keep that stuff in the job definitions/default file.